microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,252 @@
+    """,
+)
+class dpdk(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case checks DPDK can be built and installed correctly.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Sriov],
+        ),
+        priority=1,
+    )
+    def check_dpdk_build(self, case_name: str, node: Node) -> None:
+        sriov_feature = node.features[Sriov]
+        sriov_is_enabled = sriov_feature.enabled()
+        self.log.info(f"Verify SRIOV is enabled: {sriov_is_enabled}")
+        assert_that(sriov_is_enabled).is_true().described_as(
+            "SRIOV was not enabled for this test node."
+        )
+        # self._install_dpdk_dependencies(node)
+        self._hugepages_init(node)
+        self._hugepages_enable(node)
+        # self._install_dpdk(node)
+        # self._execute_expect_zero(node, "/usr/local/bin/dpdk-testpmd --version")
+        nics = self._get_nic_names(node)
+        pairings = self._get_primary_secondary_pairings(node, nics)
+
+    _ubuntu_packages = [
+        "librdmacm-dev",
+        "librdmacm1",
+        "build-essential",
+        "libnuma-dev",
+        "libmnl-dev",
+        "libelf-dev",
+        "meson",
+        "rdma-core",
+        "librdmacm-dev",
+        "librdmacm1",
+        "build-essential",
+        "libnuma-dev",
+        "libmnl-dev",
+        "libelf-dev",
+        "dpkg-dev",
+        "pkg-config",
+        "python3-pip",
+        "python3-pyelftools",
+        "python-pyelftools",
+    ]
+    _redhat_packages = [
+        "gcc",
+        "make",
+        "git",
+        "tar",
+        "wget",
+        "dos2unix",
+        "psmisc",
+        "kernel-devel",
+        "numactl-devel.x86_64",
+        "librdmacm-devel",
+        "pkgconfig",
+        "libmnl-devel",
+        "elfutils-libelf-devel",
+        "python3-pip",
+    ]
+    _rte_target = "x86_64-native-linuxapp-gcc"
+    _dpdk_github = "https://github.com/DPDK/dpdk.git"
+    _ninja_url = (
+        "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip"
+    )
+
+    _ip_addr_regex = r"[0-9]+:\s+([a-zA-Z0-9\-_]+):\s+<(?:[a-zA-Z0-9_]+,?)+>"
+
+    def _install_dpdk_dependencies(self, node: Node) -> None:
+        if isinstance(node.os, Ubuntu):
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.log.info("Packages installed for Ubunutu")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
```

Comment:
There should be a tool of dpdk, so the dependencies will be managed by the tool. Some logic is already in os and base executable class, so more code can be reused. If the tool is considered to be used by dpdk only, it can be in the same dpkd folder.

Reply:
Makes sense. will refactor